### PR TITLE
Fix some diagnostic issues with const_generics_defaults feature gate

### PR DESCRIPTION
**Pull request by BoxyUwU**
This PR makes a few changes:
- print out const param defaults in "lifetime ordering" errors rather than discarding them
- update `is_simple_text` to account for const params when checking if a type has no generics, this was causing a note to be failed to add to an error message
- fixes some diagnostic wording that incorrectly said there was ordering restrictions between type/const params despite the `const_generics_defaults` feature gate is active

### CODE REVIEW

File: compiler/rustc_middle/src/ty/diagnostics.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@ impl<'tcx> TyS<'tcx> {
     /// ADTs with no type arguments.
     pub fn is_simple_text(&self) -> bool {
         match self.kind() {
-            Adt(_, substs) => substs.types().next().is_none(),
+            Adt(_, substs) => substs.types().next().is_none() && substs.consts().next().is_none(),
```
Comment (oli-obk) [inline compiler/rustc_middle/src/ty/diagnostics.rs:57]: maybe use `non_erasable_generics`? Not that we'll be adding more stuff soon, but the point of `is_simple_text` is basically "check that there are only lifetimes in any generic list"

File: src/test/ui/const-generics/defaults/mismatch.rs [hunk 2]
```diff
@@ -17,8 +17,6 @@ fn main() {
     //~^ Error: mismatched types
     let e: Example3::<7> = ();
     //~^ Error: mismatched types
-    // FIXME(const_generics_defaults): There should be a note for the error below, but it is
-    // missing.
     let e: Example4::<7> = ();
     //~^ Error: mismatched types
```
Comment (oli-obk) [inline src/test/ui/const-generics/defaults/mismatch.rs:21]: you can test for the note without having to test for all notes by using

```suggestion
    //~^ Error: mismatched types
    //~| expected struct `Example4<7_usize>`
```

without `NOTE`

Comment (oli-obk) [reply src/test/ui/const-generics/defaults/mismatch.rs:25]: Oh, you added it to all of them... in that case you can use NOTE before the message to ensure none get lost

Comment (BoxyUwU) [reply src/test/ui/const-generics/defaults/mismatch.rs:25]: I spent a while trying to get `Note:` to work but couldn't :sweat_smile: Think ill just keep it how it is now lol the .stderr file will change if anything changes anyway so I think stuff getting dropped wouldnt be missed

**Pull request merged**